File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        maxarea=0
        left=0
        right=len(height)-1

        while left<right:
            area=(min(height[left],height[right]))*(max(left,right)-min(left,right))
            maxarea=max(maxarea,area)
            if height[left]>height[right]:
                right-=1
            else:
                left+=1

        return maxarea
        

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int low=0;
        int high=height.length-1;
        int maxvol=0;
        while(low<high){
            int vol=(high-low)*Math.min(height[low],height[high]);
            maxvol=Math.max(vol,maxvol);
            if(height[low]>height[high]){
                high--;
            }
            else{
                low++;
            }
        }
        return maxvol;
    }
}